### DISCUSSION TITLE
Is Bouncy Castle still using CRL to validate certificate validity?

### DISCUSSION THREAD

**Discussion by onepeople158** [bcgit/bc-java#1863] [Q&A]
Hello developer, while researching the Bouncy Castle API, I found that the class X509CRLObject has been enabled, and I saw the message "Do not use this class directly - either use org.bouncycastle.cert (bcpkix) or CertificateFactory." After further investigation, I did not find org.bouncycastle.cert (bcpkix). Does Bouncy Castle still have an API for CRL validation?

**Comment by dghgit** [comment]
Yes, see Java's CertPath API. The support classes (as in the public ones) are in java.security.cert.

**Reply by onepeople158** [reply]
Hello Developer,

I have used the isRevoked method in the Java library to perform a CRL (Certificate Revocation List) check, but I found that it does not return the expected result as specified by RFC 5280. Could you please let me know if Bouncy Castle or the Java library provides an API that implements the CRL validation algorithm as described in Chapter 6 of RFC 5280?

Thank you.

**Reply by dghgit** [reply]
As far as I'm aware both APIs are compliant with RFC 5280. When you say, "does not return the expected result" can you provide some details? If there's an issue with the BC API we'd certainly like to fix it, but it's impossible to tell what's going on from the current data available.

**Reply by onepeople158** [reply]
Hello, developer. The command line shows the result of my verification using OpenSSL, which displays a different range. In PyCharm, I am using the is_certificate_revoked method from the Java standard library for verification, and it does not check the CRL range according to RFC5280.
![屏幕截图 2024-11-27 084324](https://github.com/user-attachments/assets/84802bad-4df5-4c1d-860c-e2cb6afe5ca0)
![屏幕截图 2024-11-27 084908](https://github.com/user-attachments/assets/27deafcc-c109-4d11-8f6d-2a5f95a6061d)
![屏幕截图 2024-11-27 084856](https://github.com/user-attachments/assets/63906512-2fa7-480f-8c9a-b3567c44a303)

**Reply by dghgit** [reply]
I really need the certificate chain and the CRL to work out what is going on here.

**Reply by onepeople158** [reply]
Hello developer, since the is_revoked function can run without the certificate chain, I'm not sure how to add the certificate chain into the validation process.
![屏幕截图 2024-11-27 145653](https://github.com/user-attachments/assets/96cd25d7-01ab-4a1d-8339-bf8227cb46f4)
![屏幕截图 2024-11-27 144926](https://github.com/user-attachments/assets/fa63a119-18f7-47f0-816f-676d0e5d2ce6)
